rio_stac/stac.py

```diff
@@ -53,27 +56,105 @@ def get_metadata(
 
     metadata["bbox"] = list(bbox)
     metadata["footprint"] = bbox_to_geom(bbox)
+    return metadata
+
+
+def get_projection_info(
+    src_dst: Union[DatasetReader, DatasetWriter, WarpedVRT, MemoryFile]
+) -> Dict:
+    """Get projection metadata.
+
+    see: https://github.com/stac-extensions/projection/#item-properties-or-asset-fields
 
-    # Proj
+    """
     try:
         epsg = src_dst.crs.to_epsg() or None
     except AttributeError:
         epsg = None
 
-    metadata["proj"] = {
+    meta = {
         "epsg": epsg,
         "geometry": bbox_to_geom(src_dst.bounds),
         "bbox": list(src_dst.bounds),
         "shape": [src_dst.height, src_dst.width],
         "transform": list(src_dst.transform),
     }
-
     # rasterio can't reproduce wkt2
     # ref: https://github.com/mapbox/rasterio/issues/2044
     # if epsg is None:
-    #      metadata["proj"]["wkt2"] = src_dst.crs.to_wkt()
+    #      meta["wkt2"] = src_dst.crs.to_wkt()
+    return meta
 
-    return metadata
+
+def _get_stats(arr: numpy.ma.array, **kwargs: Any) -> Dict:
+    """Calculate array statistics."""
+    sample, edges = numpy.histogram(arr[~arr.mask])
+    return {
+        "statistics": {
+            "mean": arr.mean().item(),
+            "minimum": arr.min().item(),
+            "maximum": arr.max().item(),
+            "stdev": arr.std().item(),
+            "valid_percent": numpy.count_nonzero(~arr.mask)
+            / float(arr.data.size)
+            * 100,
+        },
+        "histogram": {
+            "count": len(edges),
+            "min": edges.min(),
+            "max": edges.max(),
+            "buckets": sample.tolist(),
```

Comment:
validate this https://github.com/stac-extensions/raster#histogram-object

Reply:
I think I meant to put https://github.com/developmentseed/rio-stac/pull/12#issuecomment-869072285 in reply to this. I thought you weren't using pystac, but json schema would still be an easy way to validate the output, at least in the test cases. Or you can use pystac and its extensions...

Reply:
@kylebarron when I said `validate` I meant just make sure that the values (`count` and `buckets`) were well defined, not if we were following the schema itself (which we are because we run validation in CI).